gufe/storage/externalresource/memorystorage.py

```diff
@@ -26,6 +26,9 @@ def _delete(self, location):
                 f"Unable to delete '{location}': key does not exist"
             )
 
+    def __eq__(self, other):
+        return self is other
```

Comment:
iirc this is the default object behaviour, but I guess it isn't harming anyone having it here